conans/model/version_range.py

```diff
@@ -12,7 +12,7 @@ def __init__(self, operator, version):
         self.display_version = version
 
         value = str(version)
-        if (operator == ">=" or operator == "<") and "-" not in value:
+        if (operator == ">=" or operator == "<") and "-" not in value and not version.build:
```

Comment:
Might be that the full solution is to insert ``-``, so ``1.0.0+b.2`` becomes ``1.0.0-+b.2`` (empty prerelease). Just a possibility.

Reply:
Let's talk about this some time next week when we have a tad of free time :)